astropy/io/misc/tests/test_hdf5.py

```diff
@@ -665,6 +665,7 @@ def test_read_write_unicode_to_hdf5(tmp_path):
 
 
 def assert_objects_equal(obj1, obj2, attrs, compare_class=True):
+    __tracebackhide__ = True
```

Comment:
Same again.

Note that we really should refactor to ensure these very similar functions are not repeated in multiple places!

Reply:
this one in particular has at least 2 variants (both are copy-pasted at least once). All of them are in the `io` namespace, maybe we should move them to a unified (private) test helper module (`astropy.io._testing`) ?

Reply:
`io.tests` already contains some common stuff (`mixin_columns.py`); it seems fine to add more things there (though perhaps not in this PR!).

Reply:
noted !